src/System.Management.Automation/engine/regex.cs

```diff
@@ -314,6 +314,43 @@ public static bool ContainsWildcardCharacters(string pattern)
             return result;
         }
 
+        /// <summary>
+        /// Checks if the string contains a left bracket "[" followed by a right bracket "]" after any number of characters.
+        /// </summary>
+        /// <param name="pattern"> The string to check.</param>
+        /// <returns></returns>
+        internal static bool ContainsRangeWildcard(string pattern)
```

Comment:
1. Since it is an heuristic maybe enough check only for first '['?  
2. Why not use vectorized `IndexOf()`?

Reply:
1: No, it's more user friendly to treat `[` as literal input as long as there's no ambiguity whether or not the user intended to use a wildcard.  
2: I didn't know about that. I will take a look at it later.

Reply:
It is taking escaping into account. I don't see much of an advantage in going backwards, especially not when we need to look for escape characters before each character.